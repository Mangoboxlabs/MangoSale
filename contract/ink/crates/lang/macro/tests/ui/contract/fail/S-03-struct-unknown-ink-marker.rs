use ink_lang as ink;

#[ink::contract]
mod unknown_ink_marker_on_struct {
    #[ink(storage)]
    pub struct UnknownInkMarkerOnStruct {}

    #[ink(unknown_or_unsupported)]
    pub struct HasUnknownMarker {}

    impl UnknownInkMarkerOnStruct {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
