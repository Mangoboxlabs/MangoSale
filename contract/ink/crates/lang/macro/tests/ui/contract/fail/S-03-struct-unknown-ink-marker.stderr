error: unknown ink! attribute (path)
 --> $DIR/S-03-struct-unknown-ink-marker.rs:8:11
  |
8 |     #[ink(unknown_or_unsupported)]
  |           ^^^^^^^^^^^^^^^^^^^^^^
